container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/ResourceOrProviderClassVisitor.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.container.servlet.jersey;
+
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+import javax.ws.rs.Path;
+import javax.ws.rs.ext.Provider;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * @author Tony Vaagenes
+ * @author ollivir
+ */
+public class ResourceOrProviderClassVisitor extends ClassVisitor {
+    private String className = null;
+    private boolean isPublic = false;
+    private boolean isAbstract = false;
+
+    private boolean isInnerClass = false;
+    private boolean isStatic = false;
+
+    private boolean isAnnotated = false;
+
+    public ResourceOrProviderClassVisitor() {
+        super(Opcodes.ASM5);
```

Comment:
Use ASM6 as in @gjoranv's recent PR: https://github.com/vespa-engine/vespa/pull/5919/files#diff-8.

Reply:
EDIT: Noticed that the PR was not merged. Ignore above comment.